src/Address.ts

```
@@ -5,18 +5,34 @@ import * as path from "path";
-  ].join("");
+    origin += urlObj.pathname;
+    return [
+      urlObj.protocol || "http:",
+      "//",
+      url.resolve("", origin),
+      urlObj.search || ""
+    ].join("");
+  } catch (e) {
+    // If URL is invalid, try to construct it with a base
+    const urlObj = new URL(urlRemote, "http://dummy");
+    var origin = urlObj.host || "";
```

Biome (2.4.4)

[error] Shouldn't redeclare 'origin'. Consider to delete it or rename it.

(lint/suspicious/noRedeclare)